README.md

```diff
@@ -4,10 +4,84 @@ This repository contains a Javascript implementation of [esptool](https://github
 
 `esptool-js` is based on [Web Serial API](https://wicg.github.io/serial/) and works in Google Chrome and Microsoft Edge, [version 89 or later](https://developer.mozilla.org/en-US/docs/Web/API/Serial#browser_compatibility).
 
+
+## Usage
+
+CDN
+
+``
+
+or import as npm package
+
+`npm install --save esptool-js`
+
+`yarn add --save esptool-js`
+
+Example code:
+
+```ts
+import { ESPLoader, Transport } from "esptool-js";
+
+const portFilters : { usbVendorId?: number | undefined; usbProductId?: number | undefined;}[] = [];
+const device = await navigator.serial.requestPort({ filters: portFilters });
+const transport = new Transport(device);
+
+// You can use any JavaScript compatible terminal by wrapping a helper object like this:
+let espLoaderTerminal = {
+  clean() {
+    // Implement the clean function for your terminal here.
+  },
+  writeLine(data) {
+    // Implement the writeLine function for your terminal here.
+  },
+  write(data) {
+    // Implement the write function for your terminal here.
+  },
+};
+
+const esploader: ESPLoader = new ESPLoader(transport, baudRateInteger, espLoaderTerminal);
+
+chip = await esploader.main_fn(); // Start connection with serial device, return device information string.
+
+const binaryFilesArray: {data: string; address: string;}[] = [
+  {data: "file1DataHere", address: "0x1000"}
+];
+const flash_size = "keep";
+const flash_mode = "keep";
+const flash_freq = "keep";
+const erase_all = false;
+const compress = true;
+
+// A function that will be executed on each file progress update
+const reportProgress: (fileIndex: number, written: number, total: number) => void;
+
+// A function used to do hash data verification where image is the binaryFilesArray[i].data
+const calculateMD5Hash: (image: string) => string;
+
+await esploader.write_flash(
```

Comment:
I'm slightly worried that the code in Readme will get broken, and we won't notice that.
Perhaps we could have a simple example and link to the example code from the Readme? That way at least we can test the example before the release.